### ISSUE TITLE
Clarify (and improve) testing reliability

### ISSUE THREAD

**Issue by mkilchhofer** [argoproj-labs/terraform-provider-argocd#440]
We should have a look into multiple parts of testing:
- Acceptance tests sometimes fail (relatively often)
- Against what versions of Argo CD should we test (currently we test against EOL versions of Argo CD)
- [internal/provider/data_source_application_test.go](https://github.com/argoproj-labs/terraform-provider-argocd/blob/main/internal/provider/data_source_application_test.go)  why do we specify a version constraint here? Can we remove it?
- ..? (list not complete)

**Comment by onematchfox** [comment]
> Against what versions of Argo CD should we test (currently we test against EOL versions of Argo CD)

Per https://github.com/argoproj-labs/terraform-provider-argocd?tab=readme-ov-file#compatibility-promise this should be (yes, the test config is out of date :smile:):
> This provider is compatible with at least the last 2 minor releases of ArgoCD (e.g, ranging from 1.(n).m, to 1.(n-1).0, where n is the latest available minor version).

**Comment by the-technat** [comment]
> This provider is compatible with at least the last 2 minor releases of ArgoCD (e.g, ranging from 1.(n).m, to 1.(n-1).0, where n is the latest available minor version).

So for my understanding, this would currently be 2.13 and 2.12?

**Comment by onematchfox** [comment]
> So for my understanding, this would currently be 2.12 and 2.11?

Correct. That's been the "promise" until now. Although, generally we would test the last 3 versions via the config [here](https://github.com/argoproj-labs/terraform-provider-argocd/blob/6793a99d109bad3ccfbf237134c08633f5dd8a0d/.github/workflows/tests.yml#L51).

**Comment by the-technat** [comment]
I think we should also add different Terraform versions to our test matrix, ensuring it works when not using the latest Terraform version.

**Comment by the-technat** [comment]
We already did some work on this:
- #568 
- #566 
- #565 
- #563 
- #559 

For me that could be closed.
